File: 104-maximum-depth-of-binary-tree/104-maximum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {
    const getDepth = (node, depth) => {
        if(node == null) return depth;
        let left = getDepth(node.left, depth + 1);
        let right = getDepth(node.right, depth + 1);
        return Math.max(left, right);
    }
    return getDepth(root, 0);
};

File: 104-maximum-depth-of-binary-tree/104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        
        def findDepth(node: TreeNode):
            if not node: return 0
            leftDepth = 1 + findDepth(node.left)
            rightDepth = 1 + findDepth(node.right)
            return max(leftDepth, rightDepth)
        
        return findDepth(root)